src/services/transactions.rs

```diff
@@ -37,24 +38,33 @@ pub fn get_transactions_details(tx_hash: String) -> String {
 }
 
 
-pub fn get_all_transactions(context: &Context, safe_address: &String) -> Result<Page<ServiceTransaction>> {
+pub fn get_all_transactions(context: &Context, safe_address: &String, next: &Option<String>) -> Result<Page<ServiceTransaction>> {
     let mut info_provider = InfoProvider::new(context);
     let url = format!(
-        "{}/safes/{}/all-transactions",
+        "{}/safes/{}/all-transactions/?{}",
         base_transaction_service_url(),
-        safe_address
+        safe_address,
+        next.as_ref().unwrap_or(&String::new())
     );
     let body = context.cache().request_cached(&context.client(), &url, request_cache_duration())?;
     println!("request URL: {}", &url);
+    println!("next: {:#?}", next);
     println!("{:#?}", body);
     let backend_transactions: Page<TransactionDto> = serde_json::from_str(&body)?;
     let service_transactions: Vec<ServiceTransaction> = backend_transactions.results.into_iter()
         .flat_map(|transaction| transaction.to_service_transaction(&mut info_provider).unwrap_or(vec!()))
         .collect();
     Ok(Page {
-        count: service_transactions.len(),
-        next: backend_transactions.next,
-        previous: backend_transactions.previous,
+        next: backend_transactions.next.as_ref().and_then(|link| build_paging_link(link, context)),
+        previous: backend_transactions.previous.as_ref().and_then(|link| build_paging_link(link, context)),
         results: service_transactions,
     })
+}
+
+fn build_paging_link(link: &String, context: &Context) -> Option<String> {
+    Some(format!("{}{}?next={}",
+                 context.host().unwrap(),
+                 context.path(),
```

Comment:
I would still  recommend to use the rocket macro here instead building the path ourselves. I am.also not sure if this solution perform url encoding on the next param (which it should)

Reply:
Oh yeah, sorry forgot about it, I will address it.